src/Core/Normalise/Eval.idr

```diff
@@ -394,26 +401,26 @@ parameters (defs : Defs, topopts : EvalOpts)
               Env Term free ->
               LocalEnv free args -> EvalOpts -> FC ->
               Stack free -> NF free -> List (CaseAlt args) ->
-              Core (CaseResult (NF free))
+              Core (CaseResult (TermWithEnv free))
     findAlt env loc opts fc stk val [] = do
       log "eval.casetree.stuck" 2 "Ran out of alternatives"
       pure GotStuck
     findAlt env loc opts fc stk val (x :: xs)
-         = do Result val <- tryAlt env loc opts fc stk val x
+         = do (Result (MkTermEnv newLoc val)) <- tryAlt env loc opts fc stk val x
```

Comment:
If you're not using the values and only reconstructing the closure in `pure` then I would suggest using `res@(Result {})` instead and then `pure res` later. Saves us from checking we are indeed returning the same thing.

Reply:
Thanks for the review. I made this mistake because I thought val was used (in fact, it can't be used in GotStuck branch).